system/system_error_monitor/src/system_error_monitor_core.cpp

```diff
@@ -28,6 +28,34 @@
 
 namespace
 {
+enum class DebugLevel { DEBUG = 0, INFO, WARN, ERROR };
+
+// debug: 0, info: 1, warn: 2, error: 3
+template <DebugLevel debug_level>
+void logThrottledNamed(
+  const std::string & logger_name, const rclcpp::Clock::SharedPtr clock, const double duration,
+  const std::string & message)
+{
+  static std::unordered_map<std::string, rclcpp::Time> last_output_time;
+  if (last_output_time.count(logger_name) != 0) {
+    const auto time_from_last_output = clock->now() - last_output_time.at(logger_name);
+    if (time_from_last_output.seconds() * 1000.0 < duration) {
+      return;
+    }
+  }
+
+  last_output_time[logger_name] = clock->now();
+  if constexpr (debug_level == DebugLevel::DEBUG) {
+    RCLCPP_DEBUG(rclcpp::get_logger(logger_name), message.c_str());
+  } else if constexpr (debug_level == DebugLevel::INFO) {
+    RCLCPP_INFO(rclcpp::get_logger(logger_name), message.c_str());
+  } else if constexpr (debug_level == DebugLevel::WARN) {
+    RCLCPP_WARN(rclcpp::get_logger(logger_name), message.c_str());
+  } else if constexpr (debug_level == DebugLevel::ERROR) {
+    RCLCPP_ERROR(rclcpp::get_logger(logger_name), message.c_str());
+  }
```

Comment:
It seems there is also the macro `RCLCPP_FATAL`.

https://docs.ros2.org/bouncy/api/rclcpp/logging_8hpp.html#a28cf59074387c81ef082f3e774481e95

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/510/commits/142b5a553a992dc53d0db0231db87439c646f0d9
Done